go-controller/pkg/metrics/ovn.go

```diff
@@ -342,6 +349,65 @@ func getPortCount(portType string) float64 {
 	return portCount
 }
 
+//ovnControllerSBDBConnectionCheckUpdater blocks until stopCh closed but before then polls ovn-controllers connection status with
+//southbound database periodically.
+func ovnControllerSBDBConnectionCheckUpdater(stopCh <-chan struct{}, ovsAppctl ovsClient, period time.Duration) {
+	// There maybe transient connection issues to SB DB. We want to minimise the risk of reporting this as the current state between
+	// long poll intervals.
+	retry := 5
+	sleepBetweenRetry := 5 * time.Second
+	// retry times sleepBetweenRetry
+	totalRetriesTime := 25 * time.Second
+
+	if totalRetriesTime.Seconds() >= period.Seconds() {
+		panic("period must be greater than retry total time")
+	}
+	// update metric to a good initial state
+	updateSBDBConnectionMetric(ovsAppctl, retry, sleepBetweenRetry)
+
+	ticker := time.NewTicker(period)
+	for {
+		select {
+		case <-ticker.C:
+			updateSBDBConnectionMetric(ovsAppctl, retry, sleepBetweenRetry)
+		case <-stopCh:
+			ticker.Stop()
+			return
+		}
+	}
+}
+
+func updateSBDBConnectionMetric(ovsAppctl ovsClient, retry int, sleepBetweenRetry time.Duration) {
+	var stdOut, stdErr string
+	var err error
+	var connected bool
+	connected = false
+	for i := 0; i < retry && !connected; i++ {
+		stdOut, stdErr, err = ovsAppctl("connection-status")
+		if err != nil {
+			klog.Errorf("Failed to get OVN controller southbound database connection status before utilizing"+
+				" client ovs-appctl: %v", err)
+		} else if stdErr != "" {
+			klog.Errorf("Failed to get OVN controller southbound database connection status because "+
+				"ovs-appctl command returned an error: %s", stdErr)
+		} else if stdOut == "" {
+			klog.Error("Unexpected blank output while attempting to retrieve OVN controller southbound " +
+				"database connection status")
```

Comment:
nit: inconsistent spacing `"+` vs `" +`

Reply:
ouch - my IDE (golang) is doing this every time I save it. I think it executes gofmt but oddly it leaves this inconsistent space and even when I fix it, it changes it back to inconsistent spacing.

Reply:
I didn't fix this.
Gofmt over rules my feeble attempt to change this.
This will just cause issues in the future if I attempt to fix it manually.